packages/patternfly-3/react-console/src/DesktopViewer/ConnectWithRemoteViewer.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'patternfly-react';
+
+import { VNC_CONSOLE_TYPE, SPICE_CONSOLE_TYPE } from '../common/constants';
+
+import MoreInformation from './MoreInformation';
+import MoreInformationContent from './MoreInformationContent';
+import { generateVVFile, downloadFile } from './vvFileGenerator';
+import consoleDetailPropType from './consoleDetailPropType';
+
+const ConnectWithRemoteViewer = ({
+  children,
+  onGenerate,
+  onDownload,
+  spice,
+  vnc,
+  textConnectWithRemoteViewer,
+  textMoreInfo
+}) => {
+  const console = spice || vnc; // strictly prefer spice over vnc
+  const onClick = () => {
+    const type = spice ? SPICE_CONSOLE_TYPE : VNC_CONSOLE_TYPE;
+    if (console) {
+      const vv = onGenerate({ console, type });
+      onDownload(vv.fileName || 'console.vv', vv.content, vv.mimeType || 'application/x-virt-viewer');
```

Comment:
maybe consider moving `console.vv` or `application/x-virt-viewer` to default props (and accepting another prop for those as alternates?

Reply:
I can move to constants, but the name and mime type is provided from the content generator function.

To give more insight, there's generic (default) implementation of the `.vv` file generator (see `vvFileGenerator.js`). This is good enough in most use-cases. But in particular cases (like oVirt) the content must be tailored on backend (i.e. to provide connection token). So the `onGenerate` function can be customized via props and in that case it can even choose different mime-type.

Similar case is for the actual download of the file via `onDownload`- there's default implementation working in most of the cases, but e.g. in more secured environments (tight content-security-policy) there must be different approach provided. And there are multiple possible ways, unfortunately none fitting all the cases.

Reply:
The defaults moved to constants.